app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/teidata/TEIDataPresenter.kt

```diff
@@ -367,12 +368,21 @@ class TEIDataPresenter(
     }
 
     fun enrollmentOrgUnitInCaptureScope(enrollmentOrgUnit: String): Boolean {
-        return !d2.organisationUnitModule().organisationUnits()
-            .byOrganisationUnitScope(OrganisationUnit.Scope.SCOPE_DATA_CAPTURE)
+        return !getOrganisationUnitCollectionRepository()
             .byUid().eq(enrollmentOrgUnit)
             .blockingIsEmpty()
     }
 
+    private fun getOrgListForProgram(programUid: String) =
+        getOrganisationUnitCollectionRepository()
+            .byProgramUids(listOf(programUid))
+            .blockingGet()
+
+    private fun getOrganisationUnitCollectionRepository(): OrganisationUnitCollectionRepository {
```

Comment:
In my opinion, these three previous methods (enrollmentOrgUnitInCaptureScope, enrollmentOrgUnitInCaptureScope, getOrganisationUnitCollectionRepository) should be in the repositories.

Reply:
@andresmr Fixed.